src/osd/ClassHandler.cc

```diff
@@ -313,14 +305,14 @@ void ClassHandler::ClassFilter::unregister()
 int ClassHandler::ClassMethod::exec(cls_method_context_t ctx, bufferlist& indata, bufferlist& outdata)
 {
   int ret;
-  if (cxx_func) {
+  if (func.index() == 0) {
```

Comment:
Maybe `std::get_if<type name>` instead? Visitor is also an alternative.

Reply:
actually, my last version used `std::get_if<type name>`, but i want to avoid repeating the type here. i am not sure if the additional complexity of `std::visit() and a dedicated Visitor class is worthy here.

Reply:
Personally I consider `std::get_if` and repeating the type as a lesser evil than the index-based access. IMHO it's more explicit and easier to find during a refactor. Indexing still does make an assumption on the order of `std::variant`'s parameters.

Reply:
@rzarzynski updated as suggested. PTAL!